src/App.tsx

```diff
@@ -15,36 +15,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = '',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [sortField, setSortField] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
```

Comment:
The `reset` function sets `visibleGoods` to `goodsFromServer`, but since `goodsFromServer` is a constant array, any changes to `visibleGoods` will not affect `goodsFromServer`. If `goodsFromServer` is intended to be a reference to the initial state, consider using a copy of the array instead to avoid potential side effects if the array is modified elsewhere.